calculator/src/main/java/com/market/banica/calculator/service/JMXServiceImpl.java

```diff
@@ -0,0 +1,299 @@
+        LOGGER.debug("JMX server impl: In setProductQuantity method");
+        LOGGER.info("SetProductQuantity called from JMX server with parameters recipeName {},ingredientName {}" +
+                " and newValue {}", recipeName, ingredientName, newValue);
+
+        Product parentProduct = retrieveProductFromDatabase(recipeName);
+
+        validateProductExist(recipeName, parentProduct);
+
+        int newQuantity = getValueAsInt(newValue);
+
+        if (doesIngredientBelongToRecipe(ingredientName, parentProduct)) {
+
+            setProductQuantity(recipeName, parentProduct, newQuantity);
+
+            productService.createBackUp();
+        } else {
+
+            throwExceptionWhenProductDoesNotBelongToRecipe(recipeName, ingredientName);
+        }
+
+        LOGGER.info("Value set from JMX server for recipe {} and ingredient {} with value {}"
+                , recipeName, ingredientName, newValue);
+    }
+
+    @Override
+    @ManagedOperation
+    public String getProductQuantity(String recipeName, String ingredientName) {
+        LOGGER.debug("JMX server impl: In getProductQuantity method");
+        LOGGER.info("GetProductQuantity called from JMX server for recipe {} and ingredient {}", recipeName, ingredientName);
+
+        Product parentProduct = retrieveProductFromDatabase(recipeName);
+
+        validateProductExist(recipeName, parentProduct);
+
+        if (doesIngredientBelongToRecipe(ingredientName, parentProduct)) {
+
+            LOGGER.info("Value checked from JMX server for recipe {} and ingredient {}", recipeName, ingredientName);
+            return String.valueOf(getProductQuantity(ingredientName, parentProduct));
+
+        } else {
+
+            return throwExceptionWhenProductDoesNotBelongToRecipe(recipeName, ingredientName);
+        }
+    }
+
+    @Override
+    @ManagedOperation
+    public String getUnitOfMeasure(String productName) {
+        LOGGER.debug("JMX server impl: In getUnitOfMeasure method");
+        LOGGER.info("GetUnitOfMeasure called from JMX server for product with name: {}",productName);
+
+        Product product = retrieveProductFromDatabase(productName);
+
+        validateProductExist(productName, product);
+
+        LOGGER.info("UnitOfMeasure checked from JMX server for product with name {}", productName);
+        return product.getUnitOfMeasure().toString();
+    }
+
+    @Override
+    @ManagedOperation
+    public void setUnitOfMeasure(String productName, String unitOfMeasure){
+        LOGGER.debug("JMX server impl: In setUnitOfMeasure method");
+        LOGGER.info("SetUnitOfMeasure called from JMX server for product with name: {} " +
+                "and new UnitOfMeasure: {}",productName, unitOfMeasure);
+
+        Product product = retrieveProductFromDatabase(productName);
+
+        validateProductExist(productName, product);
+
+        validateUnitOfMeasureExist(unitOfMeasure);
+
+        product.setUnitOfMeasure(Enum.valueOf(UnitOfMeasure.class,
```

Comment:
in this case, we know what is the type of the enum, so using UnitOfMeasure.valueOf is preferred

Reply:
Done